mltrace/entities/component_run.py

```diff
@@ -129,6 +143,12 @@ def set_end_timestamp(self, ts: datetime = None):
 
         self._end_timestamp = ts
 
+    def add_notes(self, notes: str):
```

Comment:
Since you already have a setter I don't think you need the `add_notes` method

Reply:
Noted. I see some attributes like `start_timestamp` have a method and others like `git_hash` have a setter method. How to decide between both please?

Reply:
Great q -- `set_start_timestamp` exists to default the `start_timestamp` to `datetime.now()` whenever the method is called. I don't think there is a `set_git_hash` in `component_run.py`